internal/controller/flow_cilium.go

```diff
@@ -0,0 +1,208 @@
+				Code: protocol.ICMPv4.Code,
+			},
+		}
+	case *flow.Layer4_ICMPv6:
+		layer4.Protocol = &pb.Layer4_Icmpv6{
+			Icmpv6: &pb.ICMPv6{
+				Type: protocol.ICMPv6.Type,
+				Code: protocol.ICMPv6.Code,
+			},
+		}
+	case *flow.Layer4_SCTP:
+		layer4.Protocol = &pb.Layer4_Sctp{
+			Sctp: &pb.SCTP{
+				SourcePort:      protocol.SCTP.SourcePort,
+				DestinationPort: protocol.SCTP.DestinationPort,
+			},
+		}
+	default:
+	}
+
+	return layer4
+}
+
+// convertCiliumWorkflows function converts a slice of flow.Workload objects to a slice of pb.Workload objects.
+func convertCiliumWorkflows(workloads []*flow.Workload) []*pb.Workload {
+	protoWorkloads := []*pb.Workload{}
+	for _, workload := range workloads {
+		protoWorkload := &pb.Workload{
+			Name: workload.GetName(),
+			Kind: workload.GetKind(),
+		}
+		protoWorkloads = append(protoWorkloads, protoWorkload)
+	}
+	return protoWorkloads
+}
+
+// convertCiliumPolicies function converts a slice of flow.Policy objects to a slice of pb.Policy objects.
+func convertCiliumPolicies(policies []*flow.Policy) []*pb.Policy {
+	protoPolicies := []*pb.Policy{}
+	for _, policy := range policies {
+		protoPolicy := &pb.Policy{
+			Name:      policy.GetName(),
+			Namespace: policy.GetNamespace(),
+			Labels:    policy.GetLabels(),
+			Revision:  policy.GetRevision(),
+		}
+		protoPolicies = append(protoPolicies, protoPolicy)
+	}
+	return protoPolicies
+}
+
+// readFlows loops continously while using the observerClient to make gRPC calls to hubble-relay and grabs the last x amount of flows.
+// It sends converts those flows to ciliumFlows then sends them over the open stream.
+func (fm *Collector) readFlows(ctx context.Context, sm streamManager) error {
+	req := &observer.GetFlowsRequest{
+		Number: flowCount,
+		Follow: true,
+	}
+	observerClient := fm.client
+	stream, err := observerClient.GetFlows(ctx, req)
+	if err != nil {
+		fm.logger.Errorw("Error getting network flows", "error", err)
+		return err
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+		flow, err := stream.Recv()
+		if err != nil {
+			fm.logger.Warnw("Failed to get flow log from stream", "error", err)
+			return err
+		}
+		ciliumFlow := createCiliumFlow(flow)
+		err = sendNetworkFlowsData(&sm, ciliumFlow)
+		if err != nil {
+			fm.logger.Errorw("Cannot send object metadata", "error", err)
+			return err
+		}
+	}
+}
+
+func createCiliumFlow(flow *observer.GetFlowsResponse) *pb.CiliumFlow {
+	flowObj := flow.GetFlow()
+	ciliumFlow := pb.CiliumFlow{
+		Time:             flowObj.GetTime(),
+		NodeName:         flowObj.GetNodeName(),
+		Verdict:          pb.Verdict(flowObj.GetVerdict()),
+		TrafficDirection: pb.TrafficDirection(flowObj.GetTrafficDirection()),
+		Layer3: &pb.IP{
+			Source:      flowObj.GetIP().GetSource(),
+			Destination: flowObj.GetIP().GetDestination(),
+			IpVersion:   pb.IPVersion(flowObj.GetIP().GetIpVersion()),
+		},
+		Layer4: convertLayer4(flowObj.GetL4()),
```

Comment:
These also may be optional.
Make sure that `convertCiliumLayer4` returns `nil` if L4 is not set.
Create a `convertCiliumIP` function to wrap the conversion of IP. Make sure that this too return `nil` if it's not set.